### PR TITLE
Add M5 template

### PR DESCRIPTION
**Pull request by steverab**
*Description of changes:*

This is an initial draft of a template for using GluonTS-based tooling to participate in the M5 forecasting competition.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by geoalgo** [comment]
Wow this is so cool!

**Comment by geoalgo** [comment]
I had a look and the notebook is very clear to me: great work!
We have this "policy" of not checking in notebooks in the format ".ipynb" but rather ".md" that are automatically converted (as the tutorial example). Maybe it would make sense to convert it this way (in a second iteration) as code reviewing and maintenance would be easier.
Also, it would be cool to add "m5" in the get_dataset so that we can get it in one line (also independent of this PR).

**Comment by steverab** [comment]
Thanks @geoalgo! I already talked to @lostella prior to submitting this PR about the `.ipynb` vs `.md` thingy and we agree that in a second iteration we should make sure that this is part of the docs and therefore under constant testing.
As for including the M5 data into the `get_dataset()` method, I am not sure whether this is possible as the data is technically only accessible on Kaggle and you need to agree to the terms of the competition before being able to download the data. This is why we currently rely on the user to just download the data from there and supply it to us in the specified directory.

**Comment by geoalgo** [comment]
Makes sense, I missed this point, thanks!

File: examples/m5_gluonts_template.ipynb [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+   "source": [
+    "cal_features = calendar.drop(['date', \n",
+    "                              'wm_yr_wk', \n",
+    "                              'weekday', \n",
+    "                              'wday', \n",
+    "                              'month', \n",
+    "                              'year', \n",
+    "                              'event_name_1', \n",
+    "                              'event_name_2', \n",
+    "                              'd'], axis=1)\n",
```
Comment (lostella) [inline examples/m5_gluonts_template.ipynb]: Other minor comment: I think we use the following newline policy elsewhere in the codebase:

```
function(
    argument1, argument2, ...
)

[
    element1, element2, ...
]
```

(essentially, going to a newline right after parentheses, and not after the first element)

Comment (steverab) [reply examples/m5_gluonts_template.ipynb]: Both comments fixed! :)

**Comment by steverab** [comment]
I agree, I'll follow up with you on this soon! :)

**Pull request merged**